static/js/components/dashboard/CourseDescription.js

```diff
@@ -18,58 +19,69 @@ export default class CourseDescription extends React.Component {
     course: Course
   };
 
-  courseDate(label: string, date: moment$Moment): string {
-    let formattedDate = date.format(DASHBOARD_FORMAT);
-    return `${label}: ${formattedDate}`;
+  needsEdxLink(run: CourseRun): boolean {
+    // Any status besides 'offered' implies that the user has at some point enrolled
+    return run.status && run.status !== STATUS_OFFERED;
   }
 
-  courseDateMessage = (firstRun: CourseRun): any => {
-    switch (firstRun.status) {
+  renderCourseDateMessage(label: string, dateString: string): string {
+    let formattedDate = moment(dateString).format(DASHBOARD_FORMAT);
+    return <span>{label}: {formattedDate}</span>;
+  }
+
+  renderDetailContents(run: CourseRun) {
+    let dateMessage, additionalDetail;
+
+    switch (run.status) {
     case STATUS_PASSED:
-      if (firstRun.course_end_date) {
-        let courseEndDate = moment(firstRun.course_end_date);
-        return this.courseDate('Ended', courseEndDate);
-      }
-      break;
     case STATUS_NOT_PASSED:
-      if (firstRun.course_end_date) {
-        let courseEndDate = moment(firstRun.course_end_date);
-        return this.courseDate('Ended', courseEndDate);
-      } else if (!_.isNil(firstRun.fuzzy_start_date)) {
-        return `Coming ${firstRun.fuzzy_start_date}`;
-      }
+      dateMessage = this.renderCourseDateMessage('Ended', run.course_end_date);
       break;
     case STATUS_CAN_UPGRADE:
     case STATUS_CURRENTLY_ENROLLED:
+    case STATUS_WILL_ATTEND:
     case STATUS_OFFERED:
-      if (firstRun.course_start_date) {
-        let courseStartDate = moment(firstRun.course_start_date);
-        return this.courseDate('Start date', courseStartDate);
+      if (run.course_start_date) {
+        dateMessage = this.renderCourseDateMessage('Start date', run.course_start_date);
+      } else if (!_.isNil(run.fuzzy_start_date)) {
+        dateMessage = <span>Coming {run.fuzzy_start_date}</span>;
       }
       break;
-    default:
-      // no runs in this course
-      return <span className="no-runs">Coming soon...</span>;
     }
 
-    return '';
-  };
+    if (run.status === STATUS_CAN_UPGRADE) {
+      additionalDetail = <span>You are Auditing this Course.</span>;
+    }
+
+    return _.compact([dateMessage, additionalDetail]);
```

Comment:
I think this is giving rise to the console error george mentioned - when returning an array of JSX objects it's necessary to add a `key` prop to each element in the array to avoid the warning.